website/frontend/src/tests/FormPayload/FormPayload.test.tsx

```diff
@@ -0,0 +1,500 @@
+  });
+
+  it('should submit correct manual', async () => {
+    render(
+      <TestFromPayload
+        values={INPUT_PAYLOAD_VALUES}
+        onSubmit={jest.fn((values: FormValues) => submitCallbackMock(values))}
+      />
+    );
+
+    toggleView();
+
+    const manualTextbox = screen.getByRole('textbox');
+
+    expect(manualTextbox).toHaveValue(getPreformattedText(INPUT_MANUAL_PAYLOAD));
+
+    const manualText = `
+    {
+      "Test1": {
+          "firstName": "first",
+          "secondName": "second",
+          "age": null,
+      }
+    }`;
+
+    changeFieldValue(manualTextbox, manualText);
+
+    expect(manualTextbox).toHaveValue(manualText);
+
+    submit();
+    await verifyValues({ payload: manualText });
+  });
+
+  it('should set form value from json', async () => {
+    const { container } = render(
+      <TestFromPayload
+        values={INPUT_PAYLOAD_VALUES}
+        onSubmit={jest.fn((values: FormValues) => submitCallbackMock(values))}
+      />
+    );
+
+    toggleView();
+
+    const textbox = screen.getByRole('textbox');
+
+    const fileInput = container.getElementsByTagName('input')[1];
+
+    expect(fileInput).toBeInTheDocument();
+    expect(screen.getByText('Select file')).toBeInTheDocument();
+
+    //upload file
+
+    const testFile = new File([INPUT_FILE_CONTENT], 'test.json', { type: FILE_TYPES.JSON });
+
+    act(() => {
+      fireEvent.change(fileInput, {
+        target: { files: [testFile] },
+      });
+    });
+
+    await waitFor(() => expect(screen.queryByText('Select file')).toBeNull());
+
+    expect(screen.getByText('test.json')).toBeInTheDocument();
+
+    await waitFor(() => expect(textbox).toHaveValue(INPUT_FILE_CONTENT));
+
+    submit();
+    await verifyValues({ payload: INPUT_FILE_CONTENT });
+
+    //drop file
+
+    act(() => {
+      fireEvent.change(fileInput, {
+        target: { files: null },
+      });
+    });
+
+    await waitFor(() => expect(screen.getByText('Select file')).toBeInTheDocument());
+
+    await waitFor(() => expect(textbox).toHaveValue(getPreformattedText(INPUT_MANUAL_PAYLOAD)));
+
+    submit();
+    await verifyValues({ payload: getPreformattedText(INPUT_MANUAL_PAYLOAD) });
+  });
+
+  it('should displayed payload', () => {
+    render(<TestFromPayload values={INPUT_PAYLOAD_VALUES} onSubmit={jest.fn()} />);
+
+    let fieldsets = screen.getAllByRole('group');
+    let selectors: HTMLSelectElement[] = screen.getAllByRole('combobox');
+
+    const changeSelectValue = (element: Element | Node, value: string) => {
+      changeFieldValue(element, value);
+
+      fieldsets = screen.getAllByRole('group');
+      selectors = screen.getAllByRole('combobox');
+    };
+
+    expect(screen.getByRole('checkbox')).not.toBeChecked();
+    expect(screen.queryByPlaceholderText('// Enter your payload here')).toBeNull();
+
+    expect(within(fieldsets[0]).getByText('Action')).toBeInTheDocument();
+    expect(selectors[0]).toHaveValue('Test1');
+
+    // Test1
```

Comment:
I'd suggest to name what type of data you testing here - enum, enum's option, result, vec, etc.